File: 백준/Bronze/27962. 오렌지먹은지오랜지/오렌지먹은지오랜지.py
N = int(input())
word = input()

for i in range(1, N):
    word1 = word[: i]
    word2 = word[-i:]

    diff = 0

    for j in range(i):
        if word1[j] != word2[j]:
            diff += 1
    
    if diff == 1:
        print('YES')
        exit(0)

print('NO')

File: 백준/Bronze/27962. 오렌지먹은지오랜지/오렌지먹은지오랜지.cc
#include <bits/stdc++.h>
using namespace std;
char str[2001];
int n,res;
int main() {
    scanf("%d",&n);
    scanf("%s",str);
    for(int i=1;i<=n;i++){
        res=0;
        for(int j=0;j<i;j++) 
            res += str[j] != str[n + j - i];
        if(res==1){
            printf("YES");
            exit(0);
        }
    }
    printf("NO");
    return 0;
}